test/finch_test.exs

```diff
@@ -276,11 +276,13 @@ defmodule FinchTest do
       assert {:ok, _} = Finch.build(:get, endpoint(bypass)) |> Finch.request(H1Finch)
 
       stop_supervised(Finch)
+    end
 
+    test "SSL - fail to select HTTP2", _bypass do
       start_supervised!({Finch, name: H2Finch, pools: %{default: [protocol: :http2]}})
 
       capture_log(fn ->
```

Comment:
This doesn't actually test anything because there's no assert. The actual log returned here is `ALPN protocol not negotiated`.

Reply:
Thanks for your reply @keathley, I have some update to make this test case more accurate (I hope so), please review and let me know if others need to tweak to keep us in the positive.

Reply:
Sorry that the missed `assert` is added now, please review again.